lnst/RecipeCommon/Perf/Measurements/XDPBenchMeasurement.py

```diff
@@ -0,0 +1,189 @@
+from lnst.RecipeCommon.Perf.Measurements.Results.AggregatedXDPBenchMeasurementResults import (
+    AggregatedXDPBenchMeasurementResults,
+)
+from lnst.Controller.RecipeResults import ResultType
+from lnst.RecipeCommon.Perf.Measurements.BaseFlowMeasurement import (
+    Flow,
+    NetworkFlowTest,
+)
+from lnst.RecipeCommon.Perf.Measurements.MeasurementError import MeasurementError
+from lnst.RecipeCommon.Perf.Measurements.Results.XDPBenchMeasurementResults import (
+    XDPBenchMeasurementResults,
+)
+from lnst.RecipeCommon.Perf.Results import (
+    PerfInterval,
+    ParallelPerfResult,
+    SequentialPerfResult,
+)
+from lnst.Tests.PktGen import PktGen
+from lnst.Tests.XDPBench import XDPBench
+from lnst.Controller.Job import Job
+from lnst.RecipeCommon.Perf.Measurements.BaseFlowMeasurement import BaseFlowMeasurement
+
+
+class XDPBenchMeasurement(BaseFlowMeasurement):
+    def __init__(self, flows: list[Flow], xdp_command: str, recipe_conf=None):
+        super().__init__(recipe_conf)
+        self._flows = flows
+        self._running_measurements = []
+        self._finished_measurements = []
+
+        self.command = xdp_command
+
+    def version(self):
+        return 1.0
+
+    @property
+    def flows(self):
+        return self._flows
+
+    def start(self):
+        net_flows = self._prepare_flows()
+        for flow in net_flows:
+            flow.server_job.start(bg=True)
+            flow.client_job.start(bg=True)
+            # server starts immediately, no need to wait
+            self._running_measurements.append(flow)
+
+        self._running_measurements = net_flows
+
+    def _prepare_server(self, flow: Flow):
+        params = {
+            "command": self.command,
+            "interface": flow.receiver_nic,
+            "duration": flow.duration + flow.warmup_duration * 2,
```

Comment:
curious... why are we multiplying the warmup duration by two?

Reply:
Current implementation of trimming cuts `warmup_duration` seconds from the start and `warmup_duration` seconds from the end of the results. So we need to extend measuring time.

See 
https://github.com/LNST-project/lnst/blob/9b220d8a13097e51e6b5ae3d533fa1fe7fc239e6/lnst/RecipeCommon/Perf/Recipe.py#L106
and `Recipe.time_aligned_results`